src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala

```diff
@@ -26,17 +26,18 @@ class HttpEventSSEHandle(request: HttpServletRequest, emitter: Emitter) extends
   lazy val id: String = UUID.randomUUID().toString
 
   private val subscribedEventTypes = request.getParameterMap.getOrDefault("event_type", Array.empty).toSet
+  private val subscribedTaskIds = request.getParameterMap.getOrDefault("task_id", Array.empty).toSet
 
-  def subscribed(eventType: String): Boolean = {
-    subscribedEventTypes.isEmpty || subscribedEventTypes.contains(eventType)
-  }
+  def subscribedForEventType(eventType: String): Boolean = subscribedEventTypes.isEmpty || subscribedEventTypes.contains(eventType)
+  def subscribedForTaskId(taskId: String): Boolean = subscribedEventTypes.isEmpty || subscribedTaskIds.contains(taskId)
 
   override def remoteAddress: String = request.getRemoteAddr
 
   override def close(): Unit = emitter.close()
 
   override def sendEvent(event: MarathonEvent): Unit = {
-    if (subscribed(event.eventType)) blocking(emitter.event(event.eventType, event.jsonString))
+    if (subscribedForEventType(event.eventType)) blocking(emitter.event(event.eventType, event.jsonString))
```

Comment:
I believe that

```
if (subscribedForEventType(event.eventType) && subscribedForTaskId(event.taskId)) blocking(emitter.event(event.eventType, event.jsonString))
```

is what you want, but since `event.taskId` is not available, you need to adjust `MarathonEvent` type and propagate task IDs throughout the entire pipeline.

Reply:
I browsed the code and tried a couple of things, but I lack the detailed knowledge of this codebase to figure out where to inject task into event flow. Any suggestions? 

Also, some of the methods that get messed up by adding taskId to MarathonEvent look like they are not designed to handle tasks, they do stuff before tasks are even created e.g ApiPostEvent. How do I deal with these? I'd suggest to use some special case of marathon event in sendEvent method. This way there's less changes to be made.

Reply:
Hi @ichernetsky , can you suggest something about the stuff I described?

Reply:
Hi @ror6ax,

I've worked on the code recently. You could do a pattern match for certain events. Checkout [Events.scala](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/core/event/Events.scala#L31) for different types.

Reply:
Btw, do you really mean task id and not instance id?